app/section/sections/mobile-memory-and-storage/ui.test.js

```diff
@@ -0,0 +1,160 @@
+    .expect(minimumConnectionSpeedQuestion.find('.nhsuk-hint').innerText).eql('Please select below the minimum memory requirement.')
+    .expect(minimumConnectionSpeedQuestion.find('select').exists).ok()
+    .expect(minimumConnectionSpeedQuestion.find('option').count).eql(8);
+});
+
+test('should populate connection speed question with existing data', async (t) => {
+  pageSetup(t, true);
+
+  const minimumConnectionSpeedQuestion = Selector('[data-test-id="combobox-options-minimum-memory-requirement"]');
+
+  await t
+    .expect(minimumConnectionSpeedQuestion.find('option[selected]').exists).ok()
+    .expect(minimumConnectionSpeedQuestion.find('option[selected]').getAttribute('value')).eql('256MB')
+});
+
+test('should render storage requirements description question', async (t) => {
+  await pageSetup(t);
+
+  const connectionRequirementDescription = Selector('[data-test-id="question-storage-requirements-description"]');
+
+  await t
+    .expect(connectionRequirementDescription.find('label.nhsuk-label').innerText).eql('Description of storage requirements')
+    .expect(connectionRequirementDescription.find('span.nhsuk-hint').innerText).eql('Add further information about the storage requirements needed for your Solution to function.')
+    .expect(connectionRequirementDescription.find('textarea').count).eql(1)
+    .expect(connectionRequirementDescription.find('[data-test-id="textarea-field-footer"]').innerText).eql('You can enter up to 300 characters');
+});
+
+test('should populate the text fields with existing data', async (t) => {
+  pageSetup(t, true);
+
+  const theQuestions = Selector('form');
+
+  await t
+    .expect(theQuestions.find('[data-test-id="question-storage-requirements-description"]').find('textarea').value).eql('Text');
+});
+
+test('should render the submit button', async (t) => {
+  await pageSetup(t);
+
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+  await t
+    .expect(submitButton.find('button').count).eql(1)
+    .expect(submitButton.find('button').innerText).eql('Save and return');
+});
+
+test('should render the return to all sections link', async (t) => {
+  await pageSetup(t);
+
+  const link = Selector('[data-test-id="section-back-link"] a');
+
+  await t
+    .expect(link.innerText).eql('Return to all sections');
+});
+
+test('should return to the marketing data dashboard when the return to all sections is clicked', async (t) => {
+  await pageSetup(t);
+
+  nock('http://localhost:8080')
+    .get('/api/v1/Solutions/S100000-001/dashboard')
+    .reply(200, dashboardWithCompleteSections);
+
+  const getLocation = ClientFunction(() => document.location.href);
+
+  const link = Selector('[data-test-id="section-back-link"]');
+
+  await t
+    .click(link.find('a'))
+    .expect(getLocation()).notContains('section')
+    .expect(getLocation()).contains('solution/S100000-001');
+});
+
```

Comment:
A few more tests to check the validation. 
Check when the "minimum-memory-requirement" returns a 400 with a required.
Check when the "storage-requirements-description" returns a 400 with a required.
Check when the "storage-requirements-description" returns a 400 with a maxLength.